File: flowsint-dossier/src/api/ownDataApi.ts
import axios from 'axios';

// Configuração da API OwnData
const API_URL = import.meta.env.VITE_OWNDATA_API_URL || 'https://completa.workbuscas.com/api';
const API_TOKEN = import.meta.env.VITE_OWNDATA_TOKEN || 'HhsbkqwwpEJIhEZPQGAPOjmC';

// Mapeamento de módulos
// CPF -> cpf, Telefone -> phone, Nome -> name, Email -> mail
// Título -> title, CEP -> cep, Mãe -> mother, CNPJ -> cnpj
type OwnDataModule = 'cpf' | 'phone' | 'name' | 'mail' | 'title' | 'cep' | 'mother' | 'cnpj';

const api = axios.create({
  baseURL: API_URL,
  timeout: 30000, // 30 segundos
});

/**
 * Interface para resposta da API OwnData
 */
export interface OwnDataResponse {
  success: boolean;
  data: any;
  message?: string;
  error?: string;
}

/**
 * Realiza uma consulta na API OwnData
 * @param module Módulo de consulta (cpf, phone, name, mail, title, cep, mother, cnpj)
 * @param query Parâmetro de busca
 */
const search = async (module: OwnDataModule, query: string): Promise<OwnDataResponse> => {
  try {
    const response = await api.get('', {
      params: {
        token: API_TOKEN,
        modulo: module,
        consulta: query.replace(/[^\w]/g, ''), // Remove caracteres especiais
      },
    });

    return {
      success: true,
      data: response.data,
    };
  } catch (error: any) {
    console.error(`Erro ao buscar ${module}:`, error);
    return {
      success: false,
      data: null,
      error: error.response?.data?.message || error.message || 'Erro ao realizar consulta',
    };
  }
};

export const ownDataApi = {
  /**
   * Busca por CPF
   * @param cpf CPF para busca (pode conter pontos e traços)
   */
  searchCpf: (cpf: string) => search('cpf', cpf),

  /**
   * Busca por Telefone
   * @param phone Telefone para busca (pode conter parênteses, espaços e traços)
   */
  searchPhone: (phone: string) => search('phone', phone),

  /**
   * Busca por Nome
   * @param name Nome completo ou parcial
   */
  searchName: (name: string) => search('name', name),

  /**
   * Busca por Email
   * @param email Email para busca
   */
  searchEmail: (email: string) => search('mail', email),

  /**
   * Busca por Título de Eleitor
   * @param title Número do título de eleitor
   */
  searchTitle: (title: string) => search('title', title),

  /**
   * Busca por CEP
   * @param cep CEP para busca (pode conter traços)
   */
  searchCep: (cep: string) => search('cep', cep),

  /**
   * Busca por nome da mãe
   * @param mother Nome da mãe
   */
  searchMother: (mother: string) => search('mother', mother),

  /**
   * Busca por CNPJ
   * @param cnpj CNPJ para busca (pode conter pontos, traços e barras)
   */
  searchCnpj: (cnpj: string) => search('cnpj', cnpj),

  /**
   * Informações sobre o token e limites
   */
  getTokenInfo: () => ({
    nome: 'LukeCyberWorm',
    valorMensal: 'R$ 450,00',
    limite: 15000,
    dataExpiracao: '26/01/2026',
    modulos: ['tel', 'cpf', 'nome', 'email', 'titulo', 'cep', 'mae', 'cnpj'],
    status: 'Ativo',
  }),
};

File: flowsint-api/app/routes/owndata_api.py
"""
OwnData API Proxy
Proxy para chamadas à API da OwnData para evitar problemas de CORS
"""
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
import httpx
import os
from typing import Optional

router = APIRouter()

OWNDATA_API_URL = os.getenv("OWNDATA_API_URL", "https://completa.workbuscas.com/api")
OWNDATA_TOKEN = os.getenv("OWNDATA_TOKEN", "HhsbkqwwpEJIhEZPQGAPOjmC")

# Request models
class CpfRequest(BaseModel):
    cpf: str

class CnpjRequest(BaseModel):
    cnpj: str

class PhoneRequest(BaseModel):
    phone: str

class EmailRequest(BaseModel):
    email: str

class NameRequest(BaseModel):
    name: str

class CepRequest(BaseModel):
    cep: str

class TitleRequest(BaseModel):
    title: str

class MotherRequest(BaseModel):
    mother: str

# Helper function to sanitize numeric inputs
def sanitize_numeric(value: str) -> str:
    """Remove non-numeric characters"""
    return ''.join(c for c in value if c.isdigit())

async def call_owndata_api(modulo: str, consulta: str) -> dict:
    """Make request to OwnData API using GET with query parameters"""
    params = {
        "token": OWNDATA_TOKEN,
        "modulo": modulo,
        "consulta": consulta
    }
    
    async with httpx.AsyncClient(timeout=30.0) as client:
        try:
            response = await client.get(OWNDATA_API_URL, params=params)
            response.raise_for_status()
            
            # Get response text
            text = response.text.strip()
            
            # If empty response, no data found
            if not text:
                return {
                    "success": True,
                    "data": {
                        "status": 404,
                        "statusMsg": "Not found",
                        "reason": "Nenhum dado encontrado na base da OwnData para esta consulta"
                    }
                }
            
            # Try to parse JSON response
            try:
                data = response.json()
                return {"success": True, "data": data}
            except:
                # If response is not valid JSON
                return {
                    "success": False,
                    "error": "Resposta inválida da API OwnData",
                    "details": text
                }
        except httpx.HTTPStatusError as e:
            return {
                "success": False,
                "error": f"Erro na API: {e.response.status_code}",
                "details": e.response.text
            }
        except Exception as e:
            return {"success": False, "error": str(e)}

@router.post("/cpf")
async def search_cpf(request: CpfRequest):
    """Buscar dados por CPF"""
    cpf = sanitize_numeric(request.cpf)
    return await call_owndata_api("cpf", cpf)

@router.post("/cnpj")
async def search_cnpj(request: CnpjRequest):
    """Buscar dados por CNPJ"""
    cnpj = sanitize_numeric(request.cnpj)
    return await call_owndata_api("cnpj", cnpj)

@router.post("/phone")
async def search_phone(request: PhoneRequest):
    """Buscar dados por Telefone"""
    phone = sanitize_numeric(request.phone)
    return await call_owndata_api("phone", phone)

@router.post("/email")
async def search_email(request: EmailRequest):
    """Buscar dados por Email"""
    return await call_owndata_api("mail", request.email.strip())

@router.post("/name")
async def search_name(request: NameRequest):
    """Buscar dados por Nome"""
    return await call_owndata_api("name", request.name.strip())

@router.post("/cep")
async def search_cep(request: CepRequest):
    """Buscar dados por CEP"""
    cep = sanitize_numeric(request.cep)
    return await call_owndata_api("cep", cep)

@router.post("/title")
async def search_title(request: TitleRequest):
    """Buscar dados por Título Eleitor"""
    title = sanitize_numeric(request.title)
    return await call_owndata_api("title", title)

@router.post("/mother")
async def search_mother(request: MotherRequest):
    """Buscar dados por Nome da Mãe"""
    return await call_owndata_api("mother", request.mother.strip())
